### ISSUE TITLE
[dev.icinga.com #10190] Problem with customvariable table update/insert queries

### ISSUE THREAD

**Issue by icinga-migration** [Icinga/icinga2#3416]
This issue has been migrated from Redmine: https://dev.icinga.com/issues/10190

**Created by mfriedrich on 2015-09-22 08:25:02 +00:00**

Assignee: _mfriedrich_
Status: _Resolved_  (closed on _2015-09-28 11:22:03 +00:00_)
Target Version: _2.3.11_
Last Update: _2015-10-16 08:21:03 +00:00 (in Redmine)_

    Icinga Version: 2.3.10
    Backport?: Already backported
    Include in Changelog: 0

---

    [2015-09-22 10:23:25 +0200] critical/IdoMysqlConnection: Error "Duplicate entry '186-1-os' for key 'object_id_2'" when executing query "UPDATE icinga_customvariablestatus SET has_been_modified = '0',  instance_id = 1,  is_json = '0',  object_id = 186,  status_update_time = FROM_UNIXTIME(1442910205),  varname = 'os',  varvalue = 'blub' WHERE object_id = 186 AND varname = 'os';INSERT INTO icinga_customvariables (config_type, has_been_modified, instance_id, is_json, object_id, varname, varvalue) VALUES ('1', '0', 1, '0', 186, 'os', 'blub');UPDATE icinga_customvariablestatus SET has_been_modified = '0',  instance_id = 1,  is_json = '0',  object_id = 186,  status_update_time = FROM_UNIXTIME(1442910205),  varname = 'os',  varvalue = 'blub' WHERE object_id = 186 AND varname = 'os';UPDATE icinga_hosts SET action_url = '',  active_checks_enabled = '1',  address = '8.8.8.8',  address6 = '',  alias = 'google.com2',  check_command_args = '',  check_command_object_id = 21,  check_interval = '5',  config_type = '1',  display_name = 'google.com2',  event_handler_enabled = '1',  failure_prediction_enabled = '0',  flap_detection_enabled = '0',  freshness_checks_enabled = '1',  freshness_threshold = '300',  high_flap_threshold = '30',  host_object_id = 186,  icon_image = '',  icon_image_alt = '',  instance_id = 1,  low_flap_threshold = '30',  max_check_attempts = '3',  notes = '',  notes_url = '',  notification_interval = '1',  notifications_enabled = '1',  notify_on_down = '0',  notify_on_downtime = '0',  notify_on_flapping = '0',  notify_on_recovery = '0',  notify_on_unreachable = '0',  obsess_over_host = '0',  passive_checks_enabled = '1',  process_performance_data = '1',  retain_nonstatus_information = '1',  retain_status_information = '1',  retry_interval = '1' WHERE host_object_id = 186;INSERT INTO icinga_host_parenthosts (host_id, instance_id, parent_host_object_id) VALUES (4, 1, 237);INSERT INTO icinga_hostdependencies (dependent_host_object_id, fail_on_down, fail_on_up, host_object_id, inherits_parent, instance_id) VALUES (186, '0', '1', 237, '1', 1);UPDATE icinga_hoststatus SET acknowledgement_type = '0',  active_checks_enabled = '1',  check_command = 'hostalive',  check_source = 'mbmif.int.netways.de',  check_type = '0',  current_check_attempt = '1',  current_notification_number = '0',  current_state = '0',  endpoint_object_id = 1,  event_handler = '',  event_handler_enabled = '1',  execution_time = '4.0322551727294922',  flap_detection_enabled = '0',  has_been_checked = '1',  host_object_id = 186,  instance_id = 1,  is_flapping = '0',  is_reachable = '1',  last_check = FROM_UNIXTIME(1442910000),  last_hard_state = '0',  last_hard_state_change = FROM_UNIXTIME(1442844903),  last_state_change = FROM_UNIXTIME(1442844845),  last_time_down = FROM_UNIXTIME(1442844554),  last_time_up = FROM_UNIXTIME(1442910000),  latency = '0',  long_output = '',  max_check_attempts = '3',  modified_host_attributes = '0',  next_check = FROM_UNIXTIME(1442910296),  normal_check_interval = '5',  notifications_enabled = '1',  output = 'PING OK - Packet loss = 0%, RTA = 4.94 ms',  passive_checks_enabled = '1',  percent_state_change = '0',  perfdata = 'rta=4.942000ms;3000.000000;5000.000000;0.000000 pl=0%;80;100;0',  problem_has_been_acknowledged = '0',  process_performance_data = '1',  retry_check_interval = '1',  scheduled_downtime_depth = '0',  should_be_scheduled = '1',  state_type = '1',  status_update_time = FROM_UNIXTIME(1442910205) WHERE host_object_id = 186;COMMIT;BEGIN"
    [2015-09-22 10:23:25 +0200] critical/IdoMysqlConnection: Exception during database operation: Verify that your database is operational!

**Changesets**

_2015-09-24 09:45:44 +00:00_ by _mfriedrich_ 8c997255a9171289c71e0db403921ebc4ccb241e

    DB IDO: Fix modified custom attribute update query problems
    
    fixes #10190

_2015-09-25 15:01:41 +00:00_ by _mfriedrich_ f1567f32230724542fb946d3efdcf10b20460890

    Disable async IDO queries for custom var updates
    
    refs #10190

_2015-09-28 11:19:42 +00:00_ by _(unknown)_ d24af1e639dfcfd37d976a8993faad551aa194a0

    Fix incorrect value for IdoMysqlConnection::m_AffectedRows
    
    refs #10190

_2015-09-28 11:24:27 +00:00_ by _mfriedrich_ f0bdd4cf21ae2335edaa32bb25817f0bc8487af9

    DB IDO: Fix modified custom attribute update query problems
    
    fixes #10190

_2015-09-28 11:25:37 +00:00_ by _(unknown)_ 39e92dfe2d713f2ccf85768db94209c36be41c98

    Fix incorrect value for IdoMysqlConnection::m_AffectedRows
    
    refs #10190

---

**Relations**:

* relates [#10190](https://dev.icinga.com/issues/10190)
* duplicates [#10190](https://dev.icinga.com/issues/10190)
